integration/test/basic/basic_test.go

```diff
@@ -85,3 +149,23 @@ func waitForReleaseStatus(gsHelmClient *helmclient.Client, release string, statu
 	b := framework.NewExponentialBackoff(framework.ShortMaxWait, framework.LongMaxInterval)
 	return backoff.RetryNotify(operation, b, notify)
 }
+
+func waitForReleaseVersion(gsHelmClient *helmclient.Client, release string, version string) error {
+	operation := func() error {
+		rh, err := gsHelmClient.GetReleaseHistory(release)
+		if err != nil {
+			return microerror.Maskf(err, "could not retrieve release history")
+		}
+		if rh.Version != version {
+			return microerror.Newf("waiting for %q, current %q", version, rh.Version)
+		}
+		return nil
+	}
+
+	notify := func(err error, t time.Duration) {
+		log.Printf("getting release version %s: %v", t, err)
+	}
+
+	b := framework.NewExponentialBackoff(framework.ShortMaxWait, framework.LongMaxInterval)
+	return backoff.RetryNotify(operation, b, notify)
```

Comment:
I would like to see errors being handled explicitly all the time. Reason is we do error masking in order to get traces. The traces tell us a story and in case we just omit certain places the story is different from real code flow and that makes debugging way harder.